### PR TITLE
feat: ingress v1

### PR DESCRIPTION
**Pull request by tokers**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

___
### Bugfix
- Description

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

___
### Backport patches
- Why need to backport?

- Source branch

- Related commits and pull requests

- Target branch

### CODE REVIEW

File: cmd/ingress/ingress_test.go [hunk 1]
```diff
@@ -105,6 +105,9 @@ func TestSignalHandler(t *testing.T) {
 }
 
 func TestNewIngressCommandEffectiveLog(t *testing.T) {
+	// Delay for 2 seconds to avoid conflicting with the
+	// TestSignalHandler case.
+	time.Sleep(2 * time.Second)
```
[the remark this answers is not in the document]
Comment (tokers) [reply cmd/ingress/ingress_test.go:110]: Not the conflicts of port, just like the conflicts about terminal (stdout).

File: docs/samples/proxy-the-httpbin-service-with-ingress.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  - host: local.httpbin.org
+    http:
+      paths:
+      - backend:
+          service:
+            name: httpbin
+            port:
+              number: 80
+        path: /
+        pathType: Prefix
```
[the remark this answers is not in the document]
Comment (tokers) [reply docs/samples/proxy-the-httpbin-service-with-ingress.md:68]: Well, it's the specification, the `pathType` can be :

* `Exact`
* `Prefix`
* `ImplementationSpecific`

File: go.mod [hunk 3]
```diff
@@ -5,6 +5,7 @@ go 1.13
 require (
 	github.com/gin-gonic/gin v1.6.3
 	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
+	github.com/gruntwork-io/terratest v0.32.3 // indirect
```
[the remark this answers is not in the document]
Comment (tokers) [reply go.mod:8]: Fixed.

[the remark this answers is not in the document]
Comment (tokers) [reply go.mod]: Fixed.

File: cmd/ingress/ingress_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tokers) [reply cmd/ingress/ingress_test.go]: Fixed by another way.

**Comment by gxthrj** [comment]
There are some conflicting files.

**Comment by tokers** [comment]
@gxthrj Fixed.

**Review by gxthrj — APPROVED** [review]

**Pull request merged**